src/index.js

```diff
@@ -34,6 +34,24 @@ function parsePart(part) {
   });
 }
 
+function validatePart(timeElementValues, validator){
+  let elements = [], isValid = true;
+
+  if(timeElementValues != null) {
+    timeElementValues.forEach((value)=> {
+      elements = elements.concat(Object.values(value))
+    });
+
+    isValid = elements.every((element)=> element >= validator.min && element <= validator.max );
+
+    if(!isValid){
+      throw new Error(
+        `Invalid time element. Range should be within [${validator.min} - ${validator.max}].`
+      );
+    }
+  }
+}
```

Comment:
I like this idea. It looks like it can be simplified:

```js
function validatePart(timeElementValues, validator) {
  const isValid = timeElementValues
    .map(timeRange => Object.values(timeRange))
    .every(element => element >= validator.min && element <= validator.max)

  if(!isValid){
    throw new Error(
      `Invalid time element. Range should be within [${validator.min} - ${validator.max}].`
    );
  }
}
```

What do you think?

Reply:
Yes this change is good, except that i would still say a null check has to be there for the case *

Reply:
And map(timeRange => Object.values(timeRange)) will return array of arrays. So it should be flattened